sphinx/texinputs/sphinxlatexobjects.sty

```diff
@@ -167,6 +183,76 @@
   \nopagebreak\noindent\kern-\labelwidth\sphinxcode{)}{#3}
   \pysigadjustitemsep
 }
+\newcommand{\pysigwithonelineperargwithonelinepertparg}[4]{
+  % #1 = name, #2 = tplist, #3 = arglist, #4 = retann
+  % render each type parameters and argument lists on its own line
+  \item[#1\sphinxcode{[}\strut]
+  \leavevmode\par\nopagebreak
+  \begingroup
+     \let\sphinxparamcomma\sphinxparamcommaoneperline
+     % \sphinxtypeparam is treated similarly to \sphinxparam but since
+     % \sphinxoptional is not accepted in a type parameters list, we do
+     % not need the hook or the global definition
+     \let\spx@sphinxtypeparam\sphinxtypeparam
+     \def\sphinxtypeparam{\def\sphinxtypeparam{\par\spx@sphinxtypeparam}\spx@sphinxtypeparam}%
+  #2\par
+  \endgroup
+  \nopagebreak\noindent\kern-\labelwidth\sphinxcode{]}
+  % render the rest of the signature like in \pysigwithonelineperarg
+  \sphinxcode{(}\strut\leavevmode\par\nopagebreak
+  \begingroup
+     \let\sphinxparamcomma\sphinxparamcommaoneperline
+     \def\sphinxoptionalhook{\ifvmode\else\kern\sphinxoptionalextraspace\relax\fi}%
+     \global\let\spx@sphinxparam\sphinxparam
+     \gdef\sphinxparam{\gdef\sphinxparam{\par\spx@sphinxparam}\spx@sphinxparam}%
+  #3\par
+  \endgroup
+  \global\let\sphinxparam\spx@sphinxparam
+  \nopagebreak\noindent\kern-\labelwidth\sphinxcode{)}{#4}
+  \pysigadjustitemsep
+}
+\newcommand{\pysiglinewithargsretwithonelinepertparg}[4]{
+  % #1 = name, #2 = tplist, #3 = arglist, #4 = retann
+  % render each type parameters on its own line but the argument list is rendered inline
+  \item[#1\sphinxcode{[}\strut]
+  \leavevmode\par\nopagebreak
+  \begingroup
+     \let\sphinxparamcomma\sphinxparamcommaoneperline
+     % \sphinxtypeparam is treated similarly to \sphinxparam but since
+     % \sphinxoptional is not accepted in a type parameters list, we do
+     % not need the hook or the global definition
+     \let\spx@sphinxtypeparam\sphinxtypeparam
+     \def\sphinxtypeparam{\def\sphinxtypeparam{\par\spx@sphinxtypeparam}\spx@sphinxtypeparam}%
+  #2\par
+  \endgroup
+  \nopagebreak\noindent\kern-\labelwidth\sphinxcode{]}
+  % render the arguments list on one line
+  \settowidth{\py@argswidth}{\sphinxcode{]}\sphinxcode{()}}%
+  \py@argswidth=\dimexpr\linewidth+\labelwidth-\py@argswidth\relax\relax
+  \sphinxcode{(}\parbox[t]{\py@argswidth}{\raggedright #3\sphinxcode{)}#4\strut}%
+  \pysigadjustitemsep
```

Comment:
ah sorry I had not seen that; probably it will need some changes. The whole business with the `\py@argswidth` and the `\parbox` was in a context when the whole thing was injected as an `\item` label.  This conditions why stuff such as `\labelwidth` is involved.  But here we are in paragraph mode, some things will have to be changed (but I lack time now).

Reply:
Yes, I think I can simply remove the `\py@argswidth` logic and write something like

```latex
\sphinxcode{(}#3\sphinxcode{)}#4\strut
```